pyaerocom/combine_vardata_ungridded.py

```diff
@@ -0,0 +1,598 @@
+        use_index = 0
+        if len(index_matches) == 0:
+            continue
+        elif len(index_matches) > 1:
+            if match_stats_how=='station_name':
+                raise Exception('Unexpected error: each station_name should '
+                                'only occur once... (perhaps due to unforeseen '
+                                'API change sometime in the future)')
+            else:
+                # more than one site was found in the surroundings of the
+                # current coordinate. Check and prefer same site name if
+                # possible, else, use closest
+                for j, idx_match in enumerate(index_matches):
+                    if statname == stats_long['station_name'][idx_match]:
+                        use_index = j
+                        break
+
+        idx_long = index_matches[use_index]
+
+        # make sure to colocate each site only once
+        if idx_long in _index_long:
+            statname_long = stats_long['station_name'][idx_long]
+            if statname == statname_long:
+                # rare case: the index match in long has already been assigned
+                # to another site in short which does not occur in long
+                # (e.g. AAOT site and Venise site in AERONET). In this case
+                # we want to use the one that matches the site name, so we
+                # have to remove the already registered index from the record
+                rm_idx = _statnames_long.index(statname_long)
+                _index_short.pop(rm_idx)
+                _index_long.pop(rm_idx)
+                _statnames_long.pop(rm_idx)
+                _statnames_short.pop(rm_idx)
+            else:
+                continue
+
+        _index_short.append(i)
+        _index_long.append(idx_long)
+        _statnames_short.append(statname)
+        _statnames_long.append(stats_long['station_name'][idx_long])
+
+    return (_index_short, _index_long, _statnames_short, _statnames_long)
+
+def _combine_2_sites(stat, var, stat_other, var_other,
+                     merge_how, merge_eval_fun,
+                     match_stats_tol_km, var_name_out,
+                     data_id_out, var_unit_out, resample_how,
+                     apply_time_resampling_constraints,
+                     min_num_obs, prefer, merge_info_vars):
+    """Combine to StationData objects for a given merge strategy
+
+    Private for now...  details should follow. Until then see
+    :func:`combine_vardata_ungridded` for details on input args
+
+    Returns
+    -------
+    StationData
+        merged StationData instance
+    """
+
+    new = StationData()
+    new.update(stat.get_meta(force_single_value=False,
+                             quality_check=False))
+
+    new.merge_meta_same_station(
+        other=stat_other,
+        check_coords=False, #has already been done
+        inplace=True,
+        raise_on_error=True) # don't raise error if meta merging fails (for now)
```

Comment:
This will make merge_meta_same_station raise an exception if error is encountered, but comment says the opposite.

Reply:
I fixed it :)